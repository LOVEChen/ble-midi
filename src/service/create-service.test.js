/* eslint-env jest */

jest.mock('bleno/lib/primary-service');

test('createService(characteristic)', () => {
  const mockPrimaryService = require('bleno/lib/primary-service');
  const characteristic = jest.fn();
  const createService = require('./create-service').default;
  const service = createService(characteristic);
  expect(mockPrimaryService.mock.calls).toMatchSnapshot();
  expect(mockPrimaryService.mock.calls.length).toEqual(1);
  const firstCall = mockPrimaryService.mock.calls[0];
  expect(firstCall[0].uuid).toEqual('03B80E5A-EDE8-4B33-A751-6CE34EC4C700');
  expect(firstCall[0].characteristics.length).toEqual(1);
  expect(firstCall[0].characteristics[0]).toEqual(characteristic);
  expect(characteristic.mock).toMatchSnapshot();
  expect(service).toBeInstanceOf(mockPrimaryService);
});
