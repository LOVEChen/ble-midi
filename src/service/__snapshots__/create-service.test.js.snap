// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createService(characteristic) 1`] = `
Object {
  "calls": Array [
    Array [
      Object {
        "characteristics": Array [
          [Function],
        ],
        "uuid": "03B80E5A-EDE8-4B33-A751-6CE34EC4C700",
      },
    ],
  ],
  "instances": Array [
    PrimaryService {
      "addListener": [Function],
      "emit": [Function],
      "eventNames": [Function],
      "getMaxListeners": [Function],
      "listenerCount": [Function],
      "listeners": [Function],
      "once": [Function],
      "prependListener": [Function],
      "prependOnceListener": [Function],
      "removeAllListeners": [Function],
      "removeListener": [Function],
      "setMaxListeners": [Function],
      "toString": [Function],
    },
  ],
}
`;

exports[`createService(characteristic) 2`] = `
Object {
  "calls": Array [],
  "instances": Array [],
}
`;

exports[`createService(characteristic) 3`] = `
PrimaryService {
  "addListener": [Function],
  "emit": [Function],
  "eventNames": [Function],
  "getMaxListeners": [Function],
  "listenerCount": [Function],
  "listeners": [Function],
  "once": [Function],
  "prependListener": [Function],
  "prependOnceListener": [Function],
  "removeAllListeners": [Function],
  "removeListener": [Function],
  "setMaxListeners": [Function],
  "toString": [Function],
}
`;
